packages/debug/src/browser/console/debug-console-contribution.tsx

```diff
@@ -78,13 +78,22 @@ export class DebugConsoleContribution extends AbstractViewContribution<ConsoleWi
                     session.on('output', event => parentConsoleSession.logOutput(parentConsoleSession.debugSession, event));
                 }
             } else {
-                const consoleSession = this.debugConsoleSessionFactory(session);
+                const consoleSession = this.debugConsoleSessionFactory(session, this.consoleSessionManager);
                 this.consoleSessionManager.add(consoleSession);
                 session.on('output', event => consoleSession.logOutput(session, event));
             }
         });
+        this.debugSessionManager.onDidChangeActiveDebugSession(session => {
+            const consoleSession = session.current ? this.consoleSessionManager.get(session.current.id) : undefined;
+            this.consoleSessionManager.selectedSession = consoleSession;
+            const consoleSelector = document.getElementById('debugConsoleSelector');
+            if (session.current && consoleSelector) {
+                (consoleSelector as HTMLSelectElement).value = session.current.id;
+            }
+        });
         this.debugSessionManager.onDidDestroyDebugSession(session => {
-            this.consoleSessionManager.delete(session.id);
+            const consoleSession = this.consoleSessionManager.delete(session.id);
+            consoleSession?.dispose();
```

Comment:
Suggestion: `consoleSessionManager` should call `dispose` on deleted console sessions. I can't envision a case where we want to delete a console session without disposing of it as well. We can revert the changes in this callback afterwards.

Reply:
You are right, I did not properly consider the parent relationship between sessions. I adapted the code to reflect that, thank you!

Reply:
Update: PR https://github.com/eclipse-theia/theia/pull/10333 actually did something similar so that is now resolved on master already.